src/v/cluster/partition.cc

```diff
@@ -820,17 +859,30 @@ ss::future<> partition::remove_remote_persistent_state(ss::abort_source& as) {
           get_ntp_config(),
           get_ntp_config().is_archival_enabled(),
           get_ntp_config().is_read_replica_mode_enabled());
-        co_await _cloud_storage_partition->erase(
+
+        if (!group_configuration().is_voter(_raft->self())) {
+            // Learners are excluded from deletion, because they have a high
+            // risk of having some out of date state, and because there will
+            // always be some voter peers to do the work.
+            co_return;
+        }
+
+        const auto finalize = co_await maybe_finalize(
           as, _raft->self(), group_configuration());
+
+        co_await _cloud_storage_partition->erase(as, finalize);
     } else if (_cloud_storage_partition && tiered_storage) {
         // Tiered storage is enabled, but deletion is disabled: ensure the
         // remote metadata is up to date before we drop the local partition.
         vlog(
           clusterlog.info,
           "Leaving tiered storage objects behind for partition {}",
           ntp());
-        co_await _cloud_storage_partition->finalize(
+        const auto finalize = co_await maybe_finalize(
           as, _raft->self(), group_configuration());
+        if (finalize) {
+            co_await _cloud_storage_partition->finalize(as);
+        }
```

Comment:
If we can square away the voter ambiguity, I think this can be rewritten to be a bit more straightforward:

```
if (not voter or not tiered storage) {
  co_return;
}
if (voter position != 0) {
  co_await sleep();
}
auto finalize_result = co_await finalize();
if (remote delete && finalize_result.success) {
  co_await erase(finalize_result.manifest); // pass the optional finalized manifest to erase()
}
```

Reply:
Definitely. I don't know the intricacies of raft, just that the goal of the PR is to preserve existing behavior. If someone knows what the differences are between how voter is checked here then I'm happy to change it.

Reply:
i think it makes sense to use `voter_position` in all the cases, this will naturally ignore all the learners and only use voters from current configuration. I have just one question. We need to remember that the leader may still be in the old configuration, is that interfere with this logic ?

Reply:
I don't think the config change edge cases cause a problem here: once the tombstone change goes in, all deletion on this path becomes best-effort and the reliable deletion happens from the (new) scrubber.  So we should lean to simplicity and use the same check in both places.

No pressure on Noah to make the change though, as this PR has nothing to do with simplifying this code :-)